### PR TITLE
feat(chat-e2e): added playwright config for overlay e2e tests

### PR DESCRIPTION
**Pull request by irinakartun**
**Description:**

added playwright config for overlay e2e tests; addded first overlay sandboxtest

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
No release type found in pull request title "fear/overlay-e2e-config: added playwright config for overlay e2e test…". Add a prefix to indicate what kind of release this pull request corresponds to. For reference, see https://www.conventionalcommits.org/

Available types:
 - fix
 - hotfix
 - feat
 - feature
 - chore
```

File: apps/chat-e2e/config/overlay.playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      detail: true,
+      outputFolder: `apps/chat-e2e/${ResultFolder.allureOverlayReport}`,
+    },
+  ],
+];
+config.use!.baseURL = 'http://127.0.0.1:4200';
+
+config.webServer = {
+  cwd: workspaceRoot,
+  command: 'npx nx serve:sandbox overlay-sandbox',
```
Comment (mikitabut) [inline apps/chat-e2e/config/overlay.playwright.config.ts]: add --configuration=production

File: apps/chat-e2e/config/playwright.config.ts
Comment (mikitabut) [inline apps/chat-e2e/config/playwright.config.ts:1]: Add chat to filename to have more explicit meaning

File: apps/chat-e2e/src/testData/expectedConstants.ts [hunk 2]
```diff
@@ -235,6 +235,7 @@ export enum Theme {
 
 export enum ResultFolder {
   allureReport = 'allure-results',
```
Comment (mikitabut) [inline apps/chat-e2e/src/testData/expectedConstants.ts]: add chat label inside to have more explicit meaning

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/overlay-sandbox/app/cases/overlay-manager/page.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { useEffect, useRef } from 'react';
 
 const overlayOptions: Omit<ChatOverlayManagerOptions, 'hostDomain'> = {
   id: 'test',
-  domain: 'http://localhost:3000',
+  domain: process.env.E2E_OVERLAY_HOST ?? 'http://localhost:3000',
```
Comment (mikitabut) [inline apps/overlay-sandbox/app/cases/overlay-manager/page.tsx]: I don't think that E2E overlay host should be insidenormal build - change it to just process.env.OVERLAY_HOST and set it on sandbox app level in env variables if possible

File: package.json [hunk 4]
```diff
@@ -14,6 +14,7 @@
     "publish:dry": "nx run-many -t publish --output-style=static --dry",
     "e2e": "nx run-many -t e2e --output-style=stream --skipInstall",
     "e2e:ci": "nx run-many -t e2e --configuration=production --output-style=stream --skipInstall",
+    "e2e:overlay:ci": "nx run-many -t e2e --configuration=productionOverlay --output-style=stream --skipInstall",
```
Comment (mikitabut) [inline package.json]: maybe it's better to extend previous command in nx to have single point of e2e starts?

File: apps/overlay-sandbox/app/cases/overlay-manager/page.tsx (hunk 3 above)
Comment (irinakartun) [reply apps/overlay-sandbox/app/cases/overlay-manager/page.tsx]: thank you, resolved

File: package.json (hunk 4 above)
Comment (irinakartun) [reply package.json]: thank you, resolved

**Review by mikitabut — DISMISSED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review

>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status:

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — DISMISSED** [review]

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
deploy-review

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/491239)
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status:

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1226.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**